shared/src/business/useCases/trialSessions/deleteTrialSessionInteractor.test.js

```diff
@@ -0,0 +1,164 @@
+const {
+  deleteTrialSessionInteractor,
+} = require('./deleteTrialSessionInteractor');
+const { User } = require('../../entities/User');
+
+const { MOCK_CASE } = require('../../../test/mockCase');
+
+const MOCK_TRIAL = {
+  caseOrder: [{ caseId: 'a54ba5a9-b37b-479d-9201-067ec6e335bb' }],
+  isCalendared: false,
+  judge: {
+    userId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+  },
+  maxCases: 100,
+  sessionType: 'Regular',
+  startDate: '2001-12-01T00:00:00.000Z',
+  term: 'Fall',
+  termYear: '2025',
+  trialLocation: 'Birmingham, AL',
+  trialSessionId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+};
+
+describe('deleteTrialSessionInteractor', () => {
+  let applicationContext;
+
+  it('throws error if user is unauthorized', async () => {
+    applicationContext = {
+      environment: { stage: 'local' },
+      getCurrentUser: () => {
+        return {
+          role: User.ROLES.petitioner,
+          userId: 'petitioner',
+        };
+      },
+      getPersistenceGateway: () => ({}),
+    };
+    await expect(
+      deleteTrialSessionInteractor({
+        applicationContext,
+        trialSessionId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+      }),
+    ).rejects.toThrow('Unauthorized');
+  });
+
+  it('throws an exception when it fails to find a trial session', async () => {
+    applicationContext = {
+      getCurrentUser: () => {
+        return new User({
+          name: 'Docket Clerk',
+          role: User.ROLES.docketClerk,
+          userId: '6805d1ab-18d0-43ec-bafb-654e83405416',
+        });
+      },
+      getPersistenceGateway: () => ({
+        getTrialSessionById: () => null,
+      }),
+    };
+
+    await expect(
+      deleteTrialSessionInteractor({
+        applicationContext,
+        trialSessionId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb',
+      }),
+    ).rejects.toThrow('trial session not found');
+  });
+
+  it('throws error when trial session start date is past now', async () => {
```

Comment:
I'm confused by this description and what's tested. You can't delete a trial session that's scheduled for 2050? How is that after the start date? Either this test needs a rethinking or I'm missing something.

Reply:
The `createISODateString` function is a mocked out current timestamp get. We always get the timestamp from a utility so that we can test easily.

Reply:
OK, I think it's the title that was confusing me. I read "past now" as "in the future," but it must mean "is in the past." "Past now" could mean either of those two things — it's just a matter of perspective. :) Would you please retitle the test to be more clear?

Reply:
Updated!